.github/workflows/style_checks.yaml

```diff
@@ -37,6 +37,11 @@ jobs:
           python -m pip install ruff pre-commit
           python -m pip list
 
+      - name: Check updates of pre-commit hooks
+        run: |
+          pre-commit autoupdate
```

Comment:
Maybe run the autoupdate line only on cron jobs? So that PRs don't get slowed down by sudden Style Check changes.

Reply:
OK. I've added the if-condition so that the autoupdate command only runs on schedule or workflow_dispatch events. It means we have to check https://github.com/GenericMappingTools/pygmt/actions/workflows/style_checks.yaml to know if there are new versions for hooks. Maybe we should automatically open a PR if there are updates?

Reply:
> Maybe we should automatically open a PR if there are updates?

Maybe we could just use pre-commit.ci as was tried in https://github.com/GenericMappingTools/pygmt/pull/3283? The automation would require quite a bit of code to set up. Ideally we could use dependabot but looking at https://github.com/dependabot/dependabot-core/issues/1524, it's still not working yet.

Reply:
I'm OK with enabling pre-commit.ci for auto update.

Reply:
Ok, add these lines to `.pre-commit-config.yaml `:

```
# https://pre-commit.ci/#configuration
ci:
  autofix_prs: false
  autoupdate_schedule: quarterly
```

Reply:
The pre-commit.ci App is also installed https://github.com/GenericMappingTools/pygmt/settings/installations.